nc/source/graphics/resources/GraphicsResources.cpp

```diff
@@ -122,14 +127,79 @@ namespace nc::graphics
         return m_accessors.at(uid);
     }
 
+    void TexturesData::AddTexture(Graphics* graphics, Texture texture)
+    {
+        auto index = m_accessors.size();
+        m_accessors.emplace(texture.uid, index);
+        m_textures.push_back(std::move(texture));
+
+        UpdateTextures(graphics);
+    }
+
+    void TexturesData::AddTextures(Graphics* graphics, std::vector<Texture> textures)
+    {
+        for (auto& texture : textures)
+        {
+            auto index = m_accessors.size();
+            m_accessors.emplace(texture.uid, index);
+            m_textures.push_back(std::move(texture));
+        }
+
+        UpdateTextures(graphics);
+    }
+
+    void TexturesData::UpdateTextures(Graphics* graphics)
+    {
+        m_sampler = graphics->GetBasePtr()->CreateTextureSampler();
+
+        std::array<vk::WriteDescriptorSet, 2> writes;
+        vk::DescriptorImageInfo samplerInfo = {};
+        samplerInfo.sampler = m_sampler.get();
+
+        if (!m_texturesInitialized)
+        {
+            m_imageInfos.reserve(m_maxTexturesCount);
+
+            for (uint32_t i = 0; i < m_maxTexturesCount; i++)
+            {
+                m_imageInfos.push_back(m_textures[0].imageInfo);
+            }
+
+            m_texturesInitialized = true;
+        }
+
+        writes[0].setDstBinding(0);
+        writes[0].setDstArrayElement(0);
+        writes[0].setDescriptorType(vk::DescriptorType::eSampler);
+        writes[0].setDescriptorCount(1);
+        writes[0].setDstSet(m_descriptorSet.get());
+        writes[0].setPBufferInfo(0);
+        writes[0].setPImageInfo(&samplerInfo);
+
+        std::transform(m_textures.cbegin(), m_textures.cend(), m_imageInfos.begin(), [](const auto& texture)
```

Comment:
Adding to the comment above about limiting the size of m_textures, the output iterator will write past the end of m_imageInfos if m_textures is larger than m_imageInfos. I know it isn't supposed to be, but I don't see anything making that guarantee. It is probably appropriate to fail in that case, but we should fail by exception, not segfault.